### PR TITLE
Add developer documentation.

### PR DESCRIPTION
**Pull request by peter-fogg**
@bderusha @clintonb could you take a look at this documentation?

### CODE REVIEW

File: docs/catalogs.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Catalogs
+========
+
+A catalog is modeled as an Elasticsearch query (see :file:`course_discovery/docs/elasticsearch.rst`) returning a
+list of courses. The ``Catalog`` model lives in :file:`course_discovery/apps/catalogs/models.py`.
```
Comment (clintonb) [inline docs/catalogs.rst:5]: Consider using `automodule`. There are examples in https://github.com/edx/edx-drf-extensions/tree/master/docs.

**Comment by bderusha** [comment]
👍 do we need to do anything special to generate the sphynx docs?

**Comment by peter-fogg** [comment]
@bderusha They should automatically build when we push this to master.

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
+API
+===
+
+The meat of the course catalog API is located in this app. ``v1`` is the current version of the API.
```
Comment (clintonb) [inline docs/api.rst]: Capitalize: Course Catalog

**Comment by clintonb** [comment]
The documentation build failed: https://readthedocs.org/projects/open-edx-course-discovery/builds/4081808/

File: docs/elasticsearch.rst [hunk 3]
```diff
@@ -24,3 +24,11 @@ created; and, the alias will be assigned to the new index.
 .. code-block:: bash
 
     $ ./manage.py install_es_indexes
+
+Query string syntax
```
Comment (clintonb) [inline docs/elasticsearch.rst]: Title case: Query String Syntax

File: docs/elasticsearch.rst [hunk 4]
```diff
@@ -24,3 +24,11 @@ created; and, the alias will be assigned to the new index.
...
+-------------------
+
+We use the query string syntax to search for courses. See `the ES documentation`_ for a guide to the query string
```
Comment (clintonb) [inline docs/elasticsearch.rst]: s/ES/Elasticsearch

File: docs/course_metadata.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+Models
+------
+
+The ``course_metadata`` contains most of the models used in the course catalog API.
+
+.. automodule:: course_discovery.apps.course_metadata.models
+   :members:
+   :undoc-members:
+
+Searching for courses
```
Comment (clintonb) [inline docs/course_metadata.rst]: Capitalize: Courses

**Comment by peter-fogg** [comment]
@clintonb changes made and build fixed.

**Comment by clintonb** [comment]
👍

**Pull request merged**